### PR TITLE
SITES-1051-Avoid page reload when hash location changes

### LINKED ISSUE #1614: [Tabs][Accordion]Avoid page reload when hash location changes
Reloading the page when the hash location changes, in order to fix Chrome's issues in scrolling to the anchor element, focusing its tab and displaying its panel, can create conflicts with the customers code.

### PR DESCRIPTION
**Pull request by comanV**
- fixes the issue existent with Chrome and related browsers, which are not implementing as expected the deep linking for the item of the container (Tabs, Accordion) component, when the URI fragment is changed. In the current behaviour, the browser scrolls to the item, but don't expand its content. This behaviour is fixed in this PR and the content is also displayed, without having to reload the page.
- fixes the issue existent with Chrome and related browsers, which are not scrolling on page reload (if URL contains URI fragment) to the item of the container component (Tabs, Accordion) component that corresponds to the deep link in the URL fragment.

Fixes #1614, #1432

### CODE REVIEW

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js [hunk 1]
```diff
@@ -502,6 +507,49 @@
...
+     * Scrolls the browser when the URI fragment is changed to the item of the container Accordion component that corresponds to the deep link in the URL fragment,
+       and displays its content.
+     * This method fixes the issue existent with Chrome and related browsers, which are just scrolling to the item without displaying its content.
+     */
+    function onHashChange() {
+        if (location.hash && location.hash !== "#") {
+            var anchorLocation = decodeURIComponent(location.hash);
+            var anchorElement = document.querySelector(anchorLocation);
+            if (anchorElement && anchorElement.classList.contains("cmp-accordion__item")) {
+                var anchorElementButton = document.querySelector(anchorLocation + "-button");
```
Comment (bpauli) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js:520]: Why not reusing `expandItem` and `collapseItem` to open and close the corresponding item?

**Review by vladbailescu — CHANGES_REQUESTED** [review]
I would rather we avoid manually going through all the steps of switching/activating the referenced panel. Instead we should either `click` the proper tab/accordion button or at least call the already existing JS methods for handling selection changes.

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js (hunk 1 above)
Comment (vladbailescu) [reply content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js:520]: Or, even bettter, `toggle` if not `getItemExpanded` (it includes datalayer calls)

File: content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/site/js/tabs.js [hunk 2]
```diff
@@ -317,6 +317,56 @@
...
+    /**
+     * Scrolls the browser when the URI fragment is changed to the item of the container Tab component that corresponds to the deep link in the URI fragment,
+       and displays its content.
+     * This method fixes the issue existent with Chrome and related browsers, which are just scrolling to the item without displaying its content.
+     */
+    function onHashChange() {
+        if (location.hash && location.hash !== "#") {
+            var anchorLocation = decodeURIComponent(location.hash);
+            var anchorElement = document.querySelector(anchorLocation);
+            if (anchorElement && anchorElement.classList.contains("cmp-tabs__tab")) {
```
Comment (vladbailescu) [inline content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/site/js/tabs.js:329]: You could use `navigateAndFocusTab`

**Review by bpauli — APPROVED** [review]

**Pull request merged**